os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -478,6 +483,11 @@ void up_assert(const uint8_t *filename, int lineno)
 	lldbg("Assertion failed at file:%s line: %d\n", filename, lineno);
 #endif
 
+	/* Print the extra arguments (if any) from ASSERT_INFO macro */
+	if (assert_info_str) {
```

Comment:
`assert_info_str` is a global variable so that this if case is always true. You should check whether first character is `\0` or not.